rts/System/FileSystem/ArchiveScanner.cpp

```diff
@@ -380,26 +380,37 @@ bool CArchiveScanner::ArchiveData::GetInfoValueBool(const std::string& key) cons
 	return false;
 }
 
-
-
-static spring::recursive_mutex scannerMutex;
-static std::atomic<uint32_t> numScannedArchives{0};
-
+static std::atomic<uint32_t> numScannedArchives{ 0 };
+static std::atomic<uint32_t> numFilesHashed{ 0 };
+static spring::recursive_mutex mtx;
 
 /*
  * CArchiveScanner
  */
 
 CArchiveScanner::CArchiveScanner()
 {
+	{
+		auto fullCachePathLC = FileSystem::GetNormalizedPath(fmt::format("{}/{}/ArchiveCache{}.lua", FileSystem::GetCwd(), FileSystem::GetCacheBaseDir(), INTERNAL_VER));
+		FileSystem::FixSlashes(fullCachePathLC);
+		StringToLowerInPlace(fullCachePathLC);
+		sha512::raw_digest rd = sha512::NULL_RAW_DIGEST;
+		sha512::hex_digest hd = sha512::NULL_HEX_DIGEST;
+		sha512::calc_digest(reinterpret_cast<const uint8_t*>(fullCachePathLC.c_str()), fullCachePathLC.size(), rd.data());
+		sha512::dump_digest(rd, hd);
+
+		ipScannerMutex = std::make_unique<InterprocessRecursiveMutex>(hd.data());
+	}
+
+	std::scoped_lock lck(*ipScannerMutex);
```

Comment:
I think I miss something/don't understand something here, you create the `ipScannerMutex` mutex, you lock it here for `ReadCache` call, but then in all other functions you are locking only global **process** mutex `mtx`, so it looks to me like the interprocess lock is not use for e.g. `WriteCache` call?